core/src/test/java/org/apache/brooklyn/core/entity/EntitiesTest.java

```diff
@@ -130,5 +129,21 @@ public void testCreateGetContainsAndRemoveTags() throws Exception {
         entity.tags().removeTag(2);
         Assert.assertEquals(entity.tags().getTags(), MutableSet.of(app));
     }
-    
+
+    @Test
+    public void testWaitFor() throws Exception {
+        entity = app.createAndManageChild(EntitySpec.create(TestEntity.class));
+        Duration timeout = Duration.ONE_MILLISECOND;
+
+        Entities.waitFor(entity, applicationIdEqualTo(app.getApplicationId()), timeout);
+
+        try {
+            Entities.waitFor(entity, applicationIdEqualTo(app.getApplicationId() + "-wrong"), timeout);
+            Assert.fail();
+
+        } catch (Exception e) {
+            // expected timeout
```

Comment:
Do an assertion on the exception so you're sure it's really what you expected, using one of the `Asserts.expectedFailureContains...` variants.